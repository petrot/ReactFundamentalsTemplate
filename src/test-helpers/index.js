export * from "./local-storage.mock";
export * from "./store.mock";
export * from "./test-wrapper.mock";
