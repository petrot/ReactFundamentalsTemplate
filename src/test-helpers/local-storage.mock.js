export const mockGetItem = jest.fn();
export const mockSetItem = jest.fn();
export const mockRemoveItem = jest.fn();

export const prepareMockLocalStorage = () => {
  Object.defineProperty(window, "localStorage", {
    value: {
      getItem: (...args) => mockGetItem(...args),
      setItem: (...args) => mockSetItem(...args),
      removeItem: (...args) => mockRemoveItem(...args),
    },
  });
};
