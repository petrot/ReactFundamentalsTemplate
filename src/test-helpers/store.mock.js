export const TEST_NAME = "Test Name";

export const mockedState = {
  user: {
    isAuth: true,
    name: TEST_NAME,
    role: "admin",
    token: "mock-token",
  },
  courses: [
    {
      title: "Course 1",
      description: "Description 1",
      duration: 5,
      authors: [
        "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
        "1c972c52-3198-4098-b6f7-799b45903199",
        "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
      ],
      creationDate: "11/03/2024",
      id: "b28fd9ad-c683-4900-bf99-38a1c5748309",
    },
  ],
  authors: [
    {
      name: "author",
      id: "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
    },
    {
      name: "author2",
      id: "1c972c52-3198-4098-b6f7-799b45903199",
    },
    {
      name: "author3",
      id: "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    },
  ],
};

export const mockedStore = {
  getState: () => mockedState,
  subscribe: jest.fn(),
  dispatch: jest.fn(),
};
