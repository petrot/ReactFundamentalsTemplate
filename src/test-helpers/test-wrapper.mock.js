import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import { mockedStore } from "./store.mock";

export const TestWrapper = ({ children, store = mockedStore }) => (
  <Provider store={store}>
    <Router>{children}</Router>
  </Provider>
);
