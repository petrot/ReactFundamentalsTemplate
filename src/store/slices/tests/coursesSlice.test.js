import { mockedState } from "../../../test-helpers";
import reducer, {
  deleteCourse,
  saveCourse,
  setCourses,
  updateCourse,
} from "../coursesSlice";

describe("coursesSlice", () => {
  it("should return the initial state", async () => {
    expect(reducer(undefined, {})).toEqual([]);
  });

  it("should handle a course list being added to an empty list", async () => {
    const previousState = [];

    expect(
      reducer(previousState, setCourses(mockedState.courses))
    ).toMatchSnapshot();
  });

  it("should handle a course being added", async () => {
    const previousState = [...mockedState.courses];
    const course = { ...mockedState.courses[0], title: "A new course" };

    expect(reducer(previousState, saveCourse(course))).toMatchSnapshot();
  });

  it("should handle a course being deleted", async () => {
    const previousState = [...mockedState.courses];

    expect(
      reducer(previousState, deleteCourse(mockedState.courses[0].id))
    ).toMatchSnapshot();
  });

  it("should handle a course being updated", async () => {
    const previousState = [...mockedState.courses];
    const course = { ...mockedState.courses[0], title: "Updateds title" };

    expect(reducer(previousState, updateCourse(course))).toMatchSnapshot();
  });
});
