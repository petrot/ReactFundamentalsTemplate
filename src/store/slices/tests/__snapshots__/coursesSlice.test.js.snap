// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coursesSlice should handle a course being added 1`] = `
Array [
  Object {
    "authors": Array [
      "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
      "1c972c52-3198-4098-b6f7-799b45903199",
      "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    ],
    "creationDate": "11/03/2024",
    "description": "Description 1",
    "duration": 5,
    "id": "b28fd9ad-c683-4900-bf99-38a1c5748309",
    "title": "Course 1",
  },
  Object {
    "authors": Array [
      "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
      "1c972c52-3198-4098-b6f7-799b45903199",
      "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    ],
    "creationDate": "11/03/2024",
    "description": "Description 1",
    "duration": 5,
    "id": "b28fd9ad-c683-4900-bf99-38a1c5748309",
    "title": "A new course",
  },
]
`;

exports[`coursesSlice should handle a course being deleted 1`] = `Array []`;

exports[`coursesSlice should handle a course being updated 1`] = `
Array [
  Object {
    "authors": Array [
      "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
      "1c972c52-3198-4098-b6f7-799b45903199",
      "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    ],
    "creationDate": "11/03/2024",
    "description": "Description 1",
    "duration": 5,
    "id": "b28fd9ad-c683-4900-bf99-38a1c5748309",
    "title": "Updateds title",
  },
]
`;

exports[`coursesSlice should handle a course list being added to an empty list 1`] = `
Array [
  Object {
    "authors": Array [
      "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
      "1c972c52-3198-4098-b6f7-799b45903199",
      "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    ],
    "creationDate": "11/03/2024",
    "description": "Description 1",
    "duration": 5,
    "id": "b28fd9ad-c683-4900-bf99-38a1c5748309",
    "title": "Course 1",
  },
]
`;
